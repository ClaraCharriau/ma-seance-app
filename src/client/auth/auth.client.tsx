import { AxiosResponse } from 'axios';
import { User } from '../../models/User';
import { axiosInstance } from '../axios.config';

const TOKEN_PATH = '/token';
const REGISTRATIONS_PATH = '/registrations';

export const checkAccountExists = async (email: string): Promise<{ exists: boolean }> => {
    return await axiosInstance.post(TOKEN_PATH, {
        email: email
    });
};

export const loginUser = async (email: string, password: string): Promise<User> => {
    return await axiosInstance.post(REGISTRATIONS_PATH, {
        email: email,
        password: password
    });
};

export const signIn = async (pseudo: string, email: string, password: string): Promise<User> => {
    return await axiosInstance.post(REGISTRATIONS_PATH, {
        pseudo: pseudo,
        email: email,
        password: password
    });
};

export const updateAccount = async (pseudo: string, email: string, password: string): Promise<User> => {
    return await axiosInstance.patch(REGISTRATIONS_PATH, {
        pseudo: pseudo,
        email: email,
        password: password
    });
};

export const deleteAccount = async (id: string): Promise<AxiosResponse<any, any>> => {
    return await axiosInstance.delete(REGISTRATIONS_PATH + '/' + id);
};
