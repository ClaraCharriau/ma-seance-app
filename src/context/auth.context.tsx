import { useLocalStorage } from '../hook/local-storage-hook/useLocalStorage';
import { User } from '../model/User';
import { createContext, useContext, useMemo, useState } from 'react';
import { UserToken } from '../model/UserToken';
import { jwtDecode } from 'jwt-decode';

/* eslint-disable */
interface IAuthContext {
    currentUser: User | null;
    setCurrentUserToken: (userToken: UserToken | null) => void;
    clearCurrentUserToken: () => void;
}

const defaultContext: IAuthContext = {
    currentUser: null,
    setCurrentUserToken: (userToken: UserToken | null) => {},
    clearCurrentUserToken: () => {}
};
/* eslint-enable */

export const AuthContext = createContext<IAuthContext>(defaultContext);

interface AuthProviderProps {
    children: JSX.Element;
}

export const AuthProvider = (props: AuthProviderProps) => {
    const { children } = props;
    const { setItem, removeItem, getItem } = useLocalStorage();

    const [user, setUser] = useState<User | null>(null);
    const [userToken, setUserToken] = useState<UserToken | null>(() => {
        const storedUserToken = getItem('maSeanceId');
        const parsedToken = storedUserToken ? JSON.parse(storedUserToken) : null;

        if (parsedToken) {
            const decodedToken: { user: User } = jwtDecode(parsedToken.access_token);
            setUser(decodedToken.user);
        }
        return parsedToken;
    });

    const setCurrentUserToken = (userToken: UserToken | null) => {
        setUserToken(userToken);
        setItem('maSeanceId', JSON.stringify(userToken));
        if (userToken) {
            const decodedToken: { user: User } = jwtDecode(userToken.access_token);
            setUser(decodedToken.user);
        }
    };

    const clearCurrentUserToken = () => {
        setUserToken(null);
        removeItem('maSeanceId');
    };

    const authContext: IAuthContext = useMemo(
        () => ({
            userToken,
            currentUser: user,
            setCurrentUserToken,
            clearCurrentUserToken
        }),
        // eslint-disable-next-line
        [user, userToken]
    );

    return <AuthContext.Provider value={authContext}>{children}</AuthContext.Provider>;
};

export const useAuthContext = () => useContext(AuthContext);
