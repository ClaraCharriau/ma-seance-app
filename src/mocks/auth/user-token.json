{
    "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMzY1OGY0MWUtNGQ1Ni00OTgwLTkzYTktNjIxZWFiMmFkM2IyIiwicHNldWRvIjoiTW9uaWNhIiwiZW1haWwiOiJtb25pY2FfZ2VsbGVyQGV4YW1wbGUuY29tIiwicGFzc3dvcmQiOiIkMmEkMDYkRW1QWEZBRldoMlJxakl0enJZbEgwLmpKSDNZdnZJaGJReS44S053Q2lKaWtrRjRpcTN6ZC4ifSwiaWF0IjoxNzEzMDM1MTg0fQ.BOAUp9KpRc_2HQ3woulPzeslaQwlrMI7vOc6Pq8nEk4"
}
