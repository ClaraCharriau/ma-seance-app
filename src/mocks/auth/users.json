{
  "id": 1,
  "pseudo": "Jane",
  "email": "test@mail.com"
}
