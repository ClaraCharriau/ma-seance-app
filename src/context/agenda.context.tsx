import { createContext, useContext, useEffect, useState } from 'react';
import { getUserAgenda } from '../client/users/user.client';
import { Showtime } from '../models/Screening';
import { useAuthContext } from './auth.context';

/* eslint-disable */
interface IAgendaContext {
    showtimes: Showtime[] | [];
}

const defaultContext: IAgendaContext = {
    showtimes: []
};
/* eslint-enable */

export const AgendaContext = createContext<IAgendaContext>(defaultContext);

interface AgendaProviderProps {
    children: JSX.Element;
}

export const AgendaProvider = (props: AgendaProviderProps) => {
    const { children } = props;
    const { currentUser } = useAuthContext();
    const [showtimes, setShowtimes] = useState<Showtime[]>([]);

    useEffect(() => {
        const getAgenda = async () => {
            if (currentUser) {
                await getUserAgenda(currentUser.id).then(response => setShowtimes(response));
            }
        };
        getAgenda();
    }, [currentUser]);

    const agendaContext: IAgendaContext = {
        showtimes
    };

    return <AgendaContext.Provider value={agendaContext}>{children}</AgendaContext.Provider>;
};

export const useAgendaContext = () => useContext(AgendaContext);
