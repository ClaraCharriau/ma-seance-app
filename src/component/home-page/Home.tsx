import { Link, NavLink } from 'react-router-dom';
import { useAuthContext } from '../../context/auth.context';
import { useFavoriteContext } from '../../context/favorite.context';
import SeeDetailsBtn from '../common/see-details-btn/SeeDetailsBtn';
import style from './Home.module.css';
import CurrentlyCarousel from './currently-carousel/CurrentlyCarousel';
import FavTheaterCarousel from './fav-theater-carousel/FavTheaterCarousel';
import NextShowtimesList from './next-showtimes-list/NextShowtimesList';
import TheaterMovieCarousel from './theater-movie-carousel/TheaterMovieCarousel';
import WatchlistCarousel from './watchlist-carousel/WatchlistCarousel';

const Home = () => {
    const { currentUser } = useAuthContext();
    const { favoriteTheaters = [] } = useFavoriteContext();

    return (
        currentUser && (
            <main className={style.homePageMain}>
                <div className={style.leftColumn}>
                    {/* CINEMAS FAVORIS */}
                    <section className={style.homeLeftSection}>
                        <div className={style.sectionTitleWrapper}>
                            <NavLink to="/fav-theaters">
                                <h2 className={style.homeSectionTitle}>
                                    <svg
                                        width="17"
                                        height="15"
                                        viewBox="0 0 17 15"
                                        fill="none"
                                        xmlns="http://www.w3.org/2000/svg"
                                    >
                                        <path
                                            d="M15.1555 1.89258L15.1574 1.89472C16.5495 3.50656 16.454 6.19924 14.8837 7.86335L14.883 7.86413L8.88362 14.2492L8.88189 14.251C8.81733 14.3205 8.74178 14.3742 8.66039 14.4104C8.57907 14.4467 8.49274 14.4652 8.40616 14.4656C8.31959 14.4661 8.23309 14.4484 8.15142 14.413C8.06969 14.3776 7.99364 14.3246 7.92842 14.2559L7.92666 14.254L1.70958 7.76598L1.70912 7.7655C1.3246 7.36529 1.02017 6.88516 0.815761 6.35376C0.611341 5.82233 0.511632 5.25173 0.523306 4.67711C0.534979 4.10249 0.657761 3.53717 0.883419 3.01591C1.10906 2.49469 1.43243 2.02931 1.83242 1.64741L1.83242 1.64741L1.83412 1.64578C3.34217 0.19229 5.85404 0.290426 7.42467 1.93654L7.42465 1.93655L7.42773 1.93972L7.9888 2.51744L8.35151 2.89091L8.71008 2.51326L9.42115 1.76436L9.4221 1.76335C9.80347 1.35947 10.2568 1.04246 10.7541 0.829862C11.2512 0.617368 11.7829 0.513162 12.3177 0.522736C12.8518 0.533578 13.3789 0.659877 13.8667 0.894775C14.3547 1.12979 14.7937 1.46891 15.1555 1.89258L15.1555 1.89258Z"
                                            fill="#FFF1A7"
                                            stroke="#FFF1A7"
                                        />
                                    </svg>
                                    Cin√©mas favoris
                                </h2>
                            </NavLink>
                            <SeeDetailsBtn text="Voir plus >" navigatePath="/fav-theaters" showIcon={false} />
                        </div>
                        <FavTheaterCarousel currentUser={currentUser} />
                    </section>
                    {/* CURRENTLY SHOWING MOVIES */}
                    <section className={style.homeLeftSectionBg}>
                        <div className={style.sectionTitleWrapper}>
                            <NavLink to="/currently">
                                <h2 className={style.homeSectionTitle}>
                                    <svg
                                        width="20"
                                        height="21"
                                        viewBox="0 0 20 21"
                                        fill="none"
                                        xmlns="http://www.w3.org/2000/svg"
                                    >
                                        <path
                                            d="M16.9831 11.3173C11.0122 10.171 10.1965 9.35536 9.05081 3.3844C9.03313 3.29243 8.98395 3.20948 8.91174 3.14984C8.83953 3.09019 8.74879 3.05756 8.65513 3.05756C8.56147 3.05756 8.47073 3.09019 8.39852 3.14984C8.32631 3.20948 8.27714 3.29243 8.25945 3.3844C7.11371 9.35536 6.2981 10.171 0.32684 11.3173C0.234886 11.335 0.151944 11.3841 0.0922899 11.4562C0.0326355 11.5284 0 11.6191 0 11.7127C0 11.8063 0.0326355 11.897 0.0922899 11.9692C0.151944 12.0413 0.234886 12.0904 0.32684 12.108C6.29807 13.2541 7.11371 14.0691 8.25945 20.0407C8.27722 20.1326 8.32643 20.2154 8.39863 20.275C8.47083 20.3346 8.56152 20.3672 8.65513 20.3672C8.74874 20.3672 8.83943 20.3346 8.91163 20.275C8.98383 20.2154 9.03304 20.1326 9.05081 20.0407C10.1965 14.0691 11.0122 13.2541 16.9831 12.108C17.075 12.0904 17.158 12.0413 17.2176 11.9692C17.2773 11.897 17.3099 11.8063 17.3099 11.7127C17.3099 11.6191 17.2773 11.5284 17.2176 11.4562C17.158 11.3841 17.075 11.335 16.9831 11.3173Z"
                                            fill="#FFF1A7"
                                        />
                                        <path
                                            d="M16.5396 6.92097C16.4459 6.92092 16.3552 6.88828 16.283 6.82864C16.2108 6.769 16.1616 6.68608 16.1439 6.59413C15.7403 4.49017 15.5099 4.25975 13.4059 3.85617C13.314 3.83848 13.231 3.7893 13.1714 3.71709C13.1117 3.64488 13.0791 3.55415 13.0791 3.46049C13.0791 3.36684 13.1117 3.2761 13.1714 3.20389C13.231 3.13168 13.314 3.0825 13.4059 3.06481C15.5102 2.66093 15.7403 2.43048 16.1439 0.326849C16.1616 0.234872 16.2107 0.151929 16.283 0.0922792C16.3552 0.0326296 16.4459 0 16.5396 0C16.6332 0 16.724 0.0326296 16.7962 0.0922792C16.8684 0.151929 16.9176 0.234872 16.9352 0.326849C17.3388 2.43048 17.5689 2.66093 19.6732 3.06481C19.7652 3.0825 19.8481 3.13168 19.9078 3.20389C19.9674 3.2761 20 3.36684 20 3.46049C20 3.55415 19.9674 3.64488 19.9078 3.71709C19.8481 3.7893 19.7652 3.83848 19.6732 3.85617C17.5693 4.25972 17.3388 4.49017 16.9352 6.59413C16.9175 6.68608 16.8683 6.769 16.7961 6.82864C16.7239 6.88828 16.6332 6.92092 16.5396 6.92097ZM15.0824 3.46049C15.4225 3.57201 15.7317 3.76198 15.9848 4.01506C16.2379 4.26814 16.428 4.57722 16.5396 4.91732C16.6512 4.57722 16.8412 4.26814 17.0943 4.01506C17.3475 3.76198 17.6566 3.57201 17.9967 3.46049C17.6567 3.34887 17.3476 3.15886 17.0945 2.90579C16.8413 2.65273 16.6513 2.3437 16.5396 2.00365C16.4279 2.3437 16.2378 2.65273 15.9847 2.90579C15.7315 3.15885 15.4225 3.34887 15.0824 3.46049Z"
                                            fill="#FFF1A7"
                                        />
                                    </svg>
                                    Sorties de la semaine
                                </h2>
                            </NavLink>
                            <SeeDetailsBtn text="Voir plus >" navigatePath="/currently" showIcon={false} />
                        </div>
                        <CurrentlyCarousel />
                    </section>

                    {/* MOVIES BY THEATERS */}
                    {favoriteTheaters.length > 0 &&
                        favoriteTheaters.map((theater, index) => (
                            <section className={style.homeLeftSection} key={index}>
                                <div className={style.sectionTitleWrapper}>
                                    <NavLink to={'/theaters/' + theater.id} state={{ theater: theater }}>
                                        <h2 className={style.homeSectionTitle}>
                                            <svg
                                                width="18"
                                                height="12"
                                                viewBox="0 0 18 12"
                                                fill="none"
                                                xmlns="http://www.w3.org/2000/svg"
                                            >
                                                <path
                                                    d="M17.8496 0.348053C17.8496 0.297104 17.8384 0.246776 17.8168 0.200623C17.7953 0.15447 17.7638 0.113619 17.7247 0.0809537C17.6856 0.0482887 17.6398 0.024607 17.5905 0.0115813C17.5413 -0.00144444 17.4898 -0.00349651 17.4396 0.00557005L16.671 0.144456C14.1147 0.605675 11.5224 0.837972 8.92481 0.838585C6.32722 0.837964 3.73487 0.605663 1.17856 0.144443L0.409936 0.00555678C0.359801 -0.00351169 0.308286 -0.00146085 0.25903 0.0115644C0.209774 0.0245895 0.163979 0.0482715 0.12488 0.0809371C0.0857808 0.113603 0.0543311 0.154455 0.0327527 0.200609C0.0111743 0.246763 -6.51236e-06 0.297091 3.42992e-09 0.34804V11.652C-7.14939e-06 11.7029 0.0111733 11.7532 0.0327514 11.7994C0.0543295 11.8455 0.0857789 11.8864 0.124878 11.9191C0.163977 11.9517 0.209773 11.9754 0.259029 11.9884C0.308285 12.0015 0.3598 12.0035 0.409936 11.9944L1.17856 11.8556C6.30126 10.93 11.5484 10.93 16.6711 11.8556L17.4397 11.9944C17.4898 12.0035 17.5413 12.0015 17.5906 11.9884C17.6398 11.9754 17.6856 11.9517 17.7247 11.9191C17.7638 11.8864 17.7953 11.8455 17.8169 11.7994C17.8385 11.7532 17.8496 11.7029 17.8496 11.652L17.8496 0.348053ZM16.5009 1.54376V10.4562C11.4869 9.59496 6.36273 9.59496 1.3487 10.4562V1.54374C6.36273 2.40508 11.4869 2.4051 16.5009 1.54376Z"
                                                    fill="#FFF1A7"
                                                />
                                            </svg>
                                            {theater.name}
                                        </h2>
                                    </NavLink>
                                    <SeeDetailsBtn
                                        text="Voir plus >"
                                        navigatePath={'/theaters/' + theater.id}
                                        showIcon={false}
                                    />
                                </div>
                                <TheaterMovieCarousel theaterId={theater.id} />
                            </section>
                        ))}
                </div>
                <div className={style.rightColumn}>
                    {/* AGENDA */}
                    <section className={style.homeRightSection}>
                        <div className={style.sectionTitleWrapper}>
                            <NavLink to="/agenda">
                                <h2 className={style.homeSectionTitle}>
                                    <svg
                                        width="19"
                                        height="19"
                                        viewBox="0 0 19 19"
                                        fill="none"
                                        xmlns="http://www.w3.org/2000/svg"
                                    >
                                        <path
                                            d="M9.5 18.9999C7.62108 18.9999 5.78435 18.4427 4.22209 17.3988C2.65982 16.355 1.44218 14.8713 0.723149 13.1354C0.0041162 11.3995 -0.184015 9.48934 0.182544 7.64652C0.549104 5.8037 1.45389 4.11097 2.78249 2.78237C4.11109 1.45377 5.80383 0.548982 7.64665 0.182422C9.48946 -0.184138 11.3996 0.00399413 13.1355 0.723027C14.8714 1.44206 16.3551 2.6597 17.399 4.22196C18.4428 5.78423 19 7.62096 19 9.49988C19 12.0194 17.9991 14.4358 16.2175 16.2174C14.4359 17.999 12.0196 18.9999 9.5 18.9999ZM9.5 1.46142C7.91015 1.46142 6.35599 1.93287 5.03407 2.81614C3.71216 3.69942 2.68185 4.95486 2.07343 6.4237C1.46502 7.89253 1.30583 9.5088 1.616 11.0681C1.92616 12.6274 2.69175 14.0597 3.81595 15.1839C4.94015 16.3081 6.37247 17.0737 7.93178 17.3839C9.49109 17.694 11.1074 17.5349 12.5762 16.9264C14.045 16.318 15.3005 15.2877 16.1837 13.9658C17.067 12.6439 17.5385 11.0897 17.5385 9.49988C17.5385 7.36795 16.6916 5.32333 15.1841 3.81583C13.6765 2.30833 11.6319 1.46142 9.5 1.46142Z"
                                            fill="#FFF1A7"
                                        />
                                        <path
                                            d="M13.5192 14.2498C13.4231 14.2504 13.3277 14.232 13.2387 14.1956C13.1497 14.1592 13.0687 14.1056 13.0004 14.0379L8.98116 10.0187C8.91343 9.9504 8.85985 9.86942 8.82348 9.78038C8.78712 9.69135 8.76868 9.59601 8.76924 9.49983V3.65368C8.76924 3.45987 8.84623 3.274 8.98328 3.13695C9.12032 2.9999 9.3062 2.92291 9.50001 2.92291C9.69382 2.92291 9.87969 2.9999 10.0167 3.13695C10.1538 3.274 10.2308 3.45987 10.2308 3.65368V9.20022L14.0381 13.0002C14.1066 13.0682 14.1609 13.149 14.198 13.238C14.2351 13.3271 14.2542 13.4226 14.2542 13.5191C14.2542 13.6155 14.2351 13.7111 14.198 13.8001C14.1609 13.8892 14.1066 13.97 14.0381 14.0379C13.9698 14.1056 13.8888 14.1592 13.7998 14.1956C13.7107 14.232 13.6154 14.2504 13.5192 14.2498Z"
                                            fill="#FFF1A7"
                                        />
                                    </svg>
                                    Prochaines s√©ances
                                </h2>
                            </NavLink>
                        </div>
                        <NextShowtimesList />
                        <Link className={style.seeDetails} to="/agenda">
                            {'Voir les prochaines s√©ances >'}
                        </Link>
                    </section>

                    {/* WATCHLIST */}
                    <section className={style.homeRightSection}>
                        <div className={style.sectionTitleWrapper}>
                            <NavLink to="/watchlist">
                                <h2 className={style.homeSectionTitle}>
                                    <svg
                                        width="20"
                                        height="18"
                                        viewBox="0 0 20 18"
                                        fill="none"
                                        xmlns="http://www.w3.org/2000/svg"
                                    >
                                        <path
                                            d="M17.3867 2.11773L17.3885 2.11987C19.0064 3.99309 18.8927 7.11402 17.0735 9.04191L17.0728 9.04269L10.2118 16.3447L10.21 16.3466C10.1298 16.4329 10.0355 16.5001 9.93337 16.5456C9.83136 16.5911 9.72281 16.6143 9.61376 16.6149C9.50471 16.6154 9.39595 16.5933 9.29351 16.5488C9.19099 16.5044 9.09606 16.4381 9.01496 16.3526L9.0132 16.3508L1.90325 8.93097L1.90279 8.93049C1.4563 8.46579 1.10303 7.90851 0.865896 7.29205C0.628754 6.67555 0.51313 6.01375 0.526667 5.34737C0.540205 4.68098 0.682591 4.02527 0.944395 3.42052C1.20618 2.8158 1.58148 2.27558 2.046 1.83206L2.04601 1.83206L2.0477 1.83043C3.80372 0.137952 6.72195 0.256567 8.54292 2.16506L8.54291 2.16508L8.54599 2.16825L9.18764 2.82894L9.55034 3.2024L9.90891 2.82476L10.7221 1.9683L10.7231 1.9673C11.1658 1.49839 11.6924 1.13009 12.2704 0.883005C12.8481 0.636028 13.4663 0.514859 14.0881 0.526002C14.7093 0.538602 15.3223 0.685502 15.8893 0.958569C16.4567 1.23176 16.9666 1.62579 17.3867 2.11773Z"
                                            stroke="#FFF1A7"
                                        />
                                    </svg>
                                    Watchlist
                                </h2>
                            </NavLink>
                        </div>
                        <WatchlistCarousel currentUser={currentUser} />
                    </section>
                </div>
            </main>
        )
    );
};

export default Home;
