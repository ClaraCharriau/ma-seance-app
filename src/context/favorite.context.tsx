import { createContext, useContext, useMemo, useState } from 'react';
import { getUserFavMovies, getUserFavTheaters } from '../client/users/user.client';
import { Movie } from '../models/Movie';
import { Theater } from '../models/Theater';
import { useAuthContext } from './auth.context';

/* eslint-disable */
interface IFavoriteContext {
    favoriteTheaters: Theater[] | [];
    favoriteMovies: Movie[] | [];
}

const defaultContext: IFavoriteContext = {
    favoriteTheaters: [],
    favoriteMovies: []
};
/* eslint-enable */

export const FavoriteContext = createContext<IFavoriteContext>(defaultContext);

interface FavoriteProviderProps {
    children: JSX.Element;
}

export const FavoriteProvider = (props: FavoriteProviderProps) => {
    const { children } = props;
    const { currentUser } = useAuthContext();
    const [favoriteTheaters, setFavoriteTheaters] = useState<Theater[]>([]);
    const [favoriteMovies, setFavoriteMovies] = useState<Movie[]>([]);

    useMemo(() => {
        const getFavorites = async () => {
            if (currentUser) {
                await Promise.allSettled([
                    getUserFavTheaters(currentUser.id).then(response => setFavoriteTheaters(response)),
                    getUserFavMovies(currentUser.id).then(response => setFavoriteMovies(response.records))
                ]);
            }
        };
        getFavorites();
    }, [currentUser]);

    const favoriteContext: IFavoriteContext = {
        favoriteTheaters,
        favoriteMovies
    };

    return <FavoriteContext.Provider value={favoriteContext}>{children}</FavoriteContext.Provider>;
};

export const useFavoriteContext = () => useContext(FavoriteContext);
