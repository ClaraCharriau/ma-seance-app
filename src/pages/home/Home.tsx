import { redirect } from 'react-router-dom';
import { useAuthContext } from '../../context/auth.context';

export const homeLoader = async () => {
    const currentUser = localStorage.getItem('user');
    if (!currentUser) {
        return redirect('/login');
    }
    return null;
};

const Home = () => {
    const { currentUser } = useAuthContext();

    return (
        <main>
            <h1>Welcome</h1>
            <p>
                You are supposed to be logged to see this page, and I know you are {currentUser && currentUser.pseudo}.
            </p>
        </main>
    );
};

export default Home;
