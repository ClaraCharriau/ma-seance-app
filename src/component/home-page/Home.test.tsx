/* eslint-disable  @typescript-eslint/no-explicit-any */
import { act, render, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthProvider } from '../../context/auth.context';
import mockTheaters from '../../mocks/users/fav-theaters.json';
import mockToken from '../../mocks/auth/user-token.json';
import Home from './Home';

describe('Home Component', () => {
    const favoriteContext = require('../../context/favorite.context');

    beforeEach(() => {
        localStorage.setItem('maSeanceId', JSON.stringify(mockToken));
    });

    it('should render home page', async () => {
        // Given
        let component: any;
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockTheaters
        });

        // When
        act(() => {
            component = render(
                <AuthProvider>
                    <Router>
                        <Home />
                    </Router>
                </AuthProvider>
            );
        });

        // Then
        await waitFor(() => {
            expect(component.getByText('C2L Saint-Germain')).toBeInTheDocument();
            expect(component.container).toMatchSnapshot();
        });
    });
});
