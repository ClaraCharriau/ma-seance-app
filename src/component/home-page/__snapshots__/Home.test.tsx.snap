// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders Home component 1`] = `<div />`;
